pasqal_cloud/__init__.py

```diff
@@ -231,6 +232,81 @@ def get_batch(self, id: str, fetch_results: bool = False) -> Batch:
         self.batches[batch.id] = batch
         return batch
 
+    def get_batches(
+        self,
+        filters: Optional[BatchFilters] = None,
+        pagination_params: Optional[PaginationParams] = None,
+    ) -> PaginatedResponse:
+        """
+        Retrieve a list of batches matching filters using a pagination system.
+
+        Batches are sorted by their creation date in descending order.
+
+        If no 'pagination_params' are provided, the first 100 batches
+        matching the query will be returned by default.
+
+        Args:
+            filters: Filters to be applied to the batches.
+            pagination_params: Pagination to be applied to the query.
+
+        Examples:
+        >>> get_batches(filters=BatchFilters(status=BatchStatus.ERROR))
+        Returns the first 100 batches with an ERROR status.
+
+        >>> Get_batches(filters=BatchFilters(status=BatchStatus.ERROR),
+                     pagination_params=PaginationParams(offset=100))
+        Returns batches 101-200 with an ERROR status.
+
+        >>> Get_batches(filters=BatchFilters(status=BatchStatus.ERROR),
+                     pagination_params=PaginationParams(offset=200))
+        Returns batches 201-300 with an ERROR status.
+
+        Returns:
+            PaginatedResponse: A class instance with two fields:
+                - total: An integer representing the total number of batches
+                         matching the filters.
+                - results: A list of batches matching the filters and pagination
+                           parameters provided.
```

Comment:
This shouldn't be documented here, there is a strong risk this becomes desynchronized with the PaginatedResponse class. 

We should just clarify this returns a PaginatedResponse instance which includes the results of the API and some pagination information